extensions/amp-story/1.0/animation-presets.js

```diff
@@ -68,232 +69,284 @@ export function setStyleForPreset(el, presetName) {
+    case 'pan-up':
+      translateY = options.translateY;
+
+      return {
+        duration: 1000,
+        easing: 'linear',
+        keyframes(dimensions) {
+          const scalingFactor = calculateTargetScalingFactor(dimensions);
+          dimensions.targetWidth *= scalingFactor;
+          dimensions.targetHeight *= scalingFactor;
 
-      return scaleAndTranslate(0, offsetY, offsetX, offsetY, scalingFactor);
-    },
-  },
-  'pan-down': {
-    duration: 1000,
-    easing: 'linear',
-    keyframes(dimensions) {
-      const scalingFactor = calculateTargetScalingFactor(dimensions);
-      dimensions.targetWidth *= scalingFactor;
-      dimensions.targetHeight *= scalingFactor;
+          const offsetX = -dimensions.targetWidth / 2;
+          const offsetY = dimensions.pageHeight - dimensions.targetHeight;
 
-      const offsetX = -dimensions.targetWidth / 2;
-      const offsetY = dimensions.pageHeight - dimensions.targetHeight;
+          return scaleAndTranslate(offsetX, offsetY, offsetX, translateY ?
+            offsetY + translateY : 0, scalingFactor);
+        },
+      };
+    case 'zoom-in':
+      let {zoomStart, zoomEnd} = options;
 
-      return scaleAndTranslate(offsetX, 0, offsetX, offsetY, scalingFactor);
-    },
-  },
-  'pan-up': {
-    duration: 1000,
-    easing: 'linear',
-    keyframes(dimensions) {
-      const scalingFactor = calculateTargetScalingFactor(dimensions);
-      dimensions.targetWidth *= scalingFactor;
-      dimensions.targetHeight *= scalingFactor;
+      if (zoomStart) {
+        userAssert(zoomEnd > zoomStart, '"zoom-end" value must be greater ' +
+        'than "zoom-start" value when using "zoom-in" animation.');
+      }
 
-      const offsetX = -dimensions.targetWidth / 2;
-      const offsetY = dimensions.pageHeight - dimensions.targetHeight;
+      return {
+        duration: 1000,
+        easing: 'linear',
+        keyframes: [
+          {transform: `scale(${zoomStart || 1}, ${zoomStart || 1})`},
+          {transform: `scale(${zoomEnd || 3}, ${zoomEnd || 3})`},
+        ],
+      };
+    case 'zoom-out':
+      zoomStart = options.zoomStart;
+      zoomEnd = options.zoomEnd;
 
-      return scaleAndTranslate(offsetX, offsetY, offsetX, 0, scalingFactor);
-    },
-  },
-  'zoom-in': {
-    duration: 1000,
-    easing: 'linear',
-    keyframes: [
-      {transform: 'scale(1,1)'},
-      {transform: 'scale(3,3)'},
-    ],
-  },
-  'zoom-out': {
-    duration: 1000,
-    easing: 'linear',
-    keyframes: [
-      {transform: 'scale(3,3)'},
-      {transform: 'scale(1,1)'},
-    ],
-  },
+      if (zoomStart) {
+        userAssert(zoomStart > zoomEnd, '"zoom-start" value must be higher ' +
+        'than "zoom-end" value when using "zoom-out" animation.');
+      }
+
+      return {
+        duration: 1000,
+        easing: 'linear',
+        keyframes: [
+          {transform: `scale(${zoomStart || 3}, ${zoomStart || 3})`},
+          {transform: `scale(${zoomEnd || 1}, ${zoomEnd || 1})`},
```

Comment:
Nit: You can do `scale(zoomEnd)` with only one parameter so you save a few bytes : )

Reply:
Done.